src/compiler/types.ts

```diff
@@ -5153,8 +5153,8 @@ export interface TypeChecker {
     /** @internal */ getRecursionIdentity(type: Type): object | undefined;
     /** @internal */ getUnmatchedProperties(source: Type, target: Type, requireOptionalProperties: boolean, matchDiscriminantProperties: boolean): IterableIterator<Symbol>;
 
-    /** @internal */ isArrayType(type: Type): boolean;
-    /** @internal */ isTupleType(type: Type): boolean;
+    isArrayType(type: Type): boolean;
+    isTupleType(type: Type): boolean;
     /** @internal */ isArrayLikeType(type: Type): boolean;

```

Comment:
> (Maybe we should export `isArrayLikeType` and invent a `isTupleLikeType` instead?)
 
I'm not sure we have a need need for an `isTupleLikeType` on our end. We have very little logic that deals with tuples [[Sourcegraph query for tuples](https://sourcegraph.com/search?q=context:global+isTupleType+repo:%5Egithub%5C.com/typescript-eslint/typescript-eslint%24+-file:%5C.d%5C.ts%24&patternType=standard&sm=1&groupBy=repo)]. Do you see a reason why we'd want that?

Reply:
The main "problem" with these functions (that we have to document in this PR) is that they don't return true for things that _extend_ from Array/tuple, e.g. TS's own `NodeArray` probably won't return true even though it is an Array.

Reply:
Maybe that doesn't happen for tuples, though.

Reply:
How much code actively extends tuple though? Is it a pattern people even really use?
Seems like it would be a suuuper niche usecase to do `type T = [1,2,3] & SomeType` (I've never seen it!).

Tuples themselves are already pretty niche, so it's going to be so rare that people do the above, probably.

As josh said - we barely touch on tuples in our rules, and even in the broader OSS ecosystem [there's only 70 usages in sourcegraph](https://sourcegraph.com/search?q=context:global+.isTupleType%28+-file:%5C.d%5C.ts%24+file:.%5Bmc%5D%3Ftsx%3F%24&patternType=standard&sm=1&groupBy=repo)..

Probably not worth it TBH, considering you'd need to build it!

Reply:
I mean specifically like:

```
type SomeTuple = [..., ..., ...]

interface SomeInterface extends SomeTuple {
    someExtraInfo: ...;
}
```

Which is analogous to our `NodeArray` type for a fixed size array.

But yes, it's not a blocker for this PR, I just was hoping to have it because it rounds out `isArrayType`+`isArrayLikeType` with `isTupleType`+`isTupleLikeType`.